var/spack/repos/builtin/packages/netlib-lapack/package.py

```diff
@@ -45,7 +53,33 @@ def install(self, spec, prefix):
 
         cmake_args.extend(std_cmake_args)
 
-        with working_dir('spack-build', create=True):
+        build_dir = 'spack-build' + ('-shared' if shared else '-static')
+        with working_dir(build_dir, create=True):
             cmake('..', *cmake_args)
             make()
             make("install")
+
+
+    def install(self, spec, prefix):
+        # Always build static libraries.
+        self.install_one(spec, prefix, False)
+
+        # Build shared libraries if requested.
+        if '+shared' in spec:
+            self.install_one(spec, prefix, True)
+
+
+    def setup_dependent_package(self, module, dspec):
+        # This is WIP for a prototype interface for virtual packages.
+        # We can update this as more builds start depending on BLAS/LAPACK.
+        libdir = find_library_path('libblas.a', self.prefix.lib64, self.prefix.lib)
+
+        self.spec.blas_static_lib   = join_path(libdir, 'libblas.a')
```

Comment:
is there a way to check that *contract* is fulfilled? If a package has `provide(blas)` and `provide(lapack)`, maybe at the end of `setup_dependent_package()` one could call some static functions of the virtual classes the package implements (i.e. blas, lapack) to check that everything is set consistently?